follower/follower_test.go

```diff
@@ -91,6 +92,14 @@ func TestMain(m *testing.M) {
 	tmpDir, _ = ioutil.TempDir("", "fllw")
 	rs := m.Run()
 	os.RemoveAll(tmpDir)
+	if rs == 0 {
+		// Followers may take 10 seconds to notice the removal.
+		time.Sleep(10 * time.Second)
+		if runtime.NumGoroutine() > 2 {
+			// Heuristic to detect leaked goroutines.
+			rs = 1
+		}
+	}
```

Comment:
Only issue is that there's no way to control the order of test cases. If I could ensure that the "leaked goroutine" test case came at the end, then that would definitely be better.

It would also need to come after deleting the temp files, to ensure the followers all expire. (But that's not hard to do, if we can control the ordering.)

This is basically a simplified version of https://github.com/golang/go/blob/master/src/net/http/main_test.go#L22-L28